lib/build.py

```diff
@@ -274,6 +268,215 @@ def build_arg_get(arg):
 
 ## Functions ##
 
+def modify(cli_):
+   # In this file, “cli” is used as a global variable
+   global cli
+   cli = cli_
+
+   # This file assumes that global cli comes from the “build” function. If we
+   # don’t assign these values, the program will fail after trying to access
+   # them.
+   cli.parse_only = False
+   cli.force = ch.Force_Mode.SECCOMP
+   cli.force_cmd = force.FORCE_CMD_DEFAULT
+   cli.bind = []
+   cli.context = os.path.abspath(os.sep)
+
+   commands = []
+   # “Flatten” commands array
+   for c in cli.c:
+      commands += c
+   src_image = im.Image(im.Reference(cli.image_ref))
+   out_image = im.Image(im.Reference(cli.out_image))
+   if (not src_image.unpack_exist_p):
+      ch.FATAL("not in storage: %s" % src_image.ref)
+   if (cli.out_image == cli.image_ref):
+      ch.FATAL("output must be different from source image (%s)" % cli.image_ref)
+   if (cli.script is not None):
+      if (not ch.Path(cli.script).exists):
+         ch.FATAL("%s: no such file" % cli.script)
+
+   # This kludge is necessary because cli is a global variable, with cli.tag
+   # assumed present elsewhere in the file. Here, cli.tag represents the
+   # destination image.
+   cli.tag = str(out_image)
+
+   stdin = sys.stdin.read()
+
+   # We check that stdin isn’t None to ensure that we don’t go down this code
+   # path by mistake (e.g. in CI, where stdin will never by a TTY).
+   if ((not sys.stdin.isatty()) and (commands == []) and (stdin != '')):
+      # https://stackoverflow.com/a/6482200
+
+      # We use “decode("utf-8")” here because stdout seems default to a bytes
+      # object, which is not a valid type for an argument for “Path”.
+      tmpfile = ch.Path(subprocess.run(["mktemp", "-d"],capture_output=True).stdout.decode("utf-8"))
+      with open(tmpfile, "w") as outfile:
+         outfile.write(stdin)
+      # By default, the file is seemingly created with its execute bit
+      # unassigned. This is problematic for the RUN instruction.
```

Comment:
The script doesn’t actually have to be executable if it’s a file argument to the shell.